api/src/integrations/core/datasource.js

```diff
@@ -313,6 +317,60 @@ class CoreDataSource {
       }
     }
   }
+
+  /**
+   * Gets valid permit for a Core mine. A valid permit must meet the following criteria:
+   *  - Must not be exploratory
+   *  - Must not be historical
+   * .
+   * 
+   * @param {object} coreRecord Record from the Core API.
+   * @returns {object} Valid permit.
+   * @memberof CoreDataSource
+   */
+  async getMinePermit(coreRecord) {
+    if (!coreRecord) {
+      throw Error('getMinePermit - required coreRecord is null.');
+    }
+
+
+    try {
+      // Get permits with detailed information.
+      const url = this.getIntegrationUrl(CORE_API_HOST, `/api/mines/${coreRecord.mine_guid}/permits`);
+      const { records: permits } = await integrationUtils.getRecords(url, this.getAuthHeader());
+
+
+      // First, any mines with 'X' as their second character are considered exploratory. Remove them.
+      const nonExploratoryPermits = permits.filter(permit => permit.permit_no[1].toLowerCase() !== 'x');
+
+      // Second, mine must not be historical which is indicated by an authorized year of '9999' on the latest amendment.
+      const validPermitNumbers = []
+      for (const permit of nonExploratoryPermits) {
+        // Confirm that the most recent amendment is not historical, which is always the first index.
+        // If 'null' then it is considered valid.
+        if (!permit.permit_amendments[0].authorization_end_date) {
```

Comment:
Might be worth a null check if `permit_amendments` might be null or empty? I assume a null/empty permit_amendments set would be valid regardless?

Reply:
To get here if has to have them, but I did add a check just to be safe.